### PR TITLE
Add support for background activity start mode (Android 14)

### PR DESCRIPTION
**Pull request by cketti**
See https://developer.android.com/about/versions/14/behavior-changes-14#background-activity-restrictions

### CODE REVIEW

File: app/ui/legacy/src/main/java/com/fsck/k9/ui/crypto/MessageCryptoCallback.java [hunk 1]
```diff
@@ -10,6 +10,5 @@
 public interface MessageCryptoCallback {
```
Comment (wmontwe) [inline app/ui/legacy/src/main/java/com/fsck/k9/ui/crypto/MessageCryptoCallback.java:10]: `import android.content.Intent;` could be removed

File: app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewFragment.kt [hunk 2]
```diff
@@ -563,17 +564,15 @@ class MessageViewFragment :
         startActivityForResult(intent, requestCode)
     }
 
-    fun onPendingIntentResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```
Comment (wmontwe) [inline app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewFragment.kt:567]: This should be annotated as deprecated

File: plugins/openpgp-api-lib/openpgp-api/src/main/java/org/openintents/openpgp/util/OpenPgpIntentStarter.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            options,
+        )
+    }
+
+    @JvmStatic
+    @Throws(IntentSender.SendIntentException::class)
+    fun startIntentSenderForResult(fragment: Fragment, intentSender: IntentSender, requestCode: Int) {
+        val options = buildOptionsBundle()
+
+        @Suppress("DEPRECATION")
```
Comment (wmontwe) [inline plugins/openpgp-api-lib/openpgp-api/src/main/java/org/openintents/openpgp/util/OpenPgpIntentStarter.kt:53]: We should create an issue to migrate to the Activity Result API.

**Review by wmontwe — APPROVED** [review]
:+1:

I think we need to start migrating to the new Activity Result API but looking at the code base, this is quite some effort.

File: plugins/openpgp-api-lib/openpgp-api/src/main/java/org/openintents/openpgp/util/OpenPgpIntentStarter.kt (hunk 3 above)
Comment (cketti) [reply plugins/openpgp-api-lib/openpgp-api/src/main/java/org/openintents/openpgp/util/OpenPgpIntentStarter.kt:53]: Done. https://github.com/thunderbird/thunderbird-android/issues/7758

File: app/ui/legacy/src/main/java/com/fsck/k9/ui/crypto/MessageCryptoCallback.java (hunk 1 above)
Comment (cketti) [reply app/ui/legacy/src/main/java/com/fsck/k9/ui/crypto/MessageCryptoCallback.java:9]: Done.

File: app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewFragment.kt (hunk 2 above)
Comment (cketti) [reply app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewFragment.kt:568]: Done.

**Pull request merged**